Library/Homebrew/extend/os/linux/diagnostic.rb

```diff
@@ -0,0 +1,60 @@
+require "tempfile"
+
+module Homebrew
+  module Diagnostic
+    class Checks
+      alias mac_message check_tmpdir_sticky_bit
+      def check_tmpdir_sticky_bit
+        mac_message + <<~EOS
+          If you don't have administrative privileges on this machine,
+          create a directory and point HOMEBREW_TEMP environment variable to it,
+          for example:
+            HOMEBREW_TEMP="$HOME"/tmp
+            mkdir -p "$HOMEBREW_TEMP"
+            chmod 1755 "$HOMEBREW_TEMP"
+            echo "export HOMEBREW_TEMP=\"$HOMEBREW_TEMP\"" >> #{shell_profile}
+        EOS
+      end
+
+      def check_xdg_data_dirs
+        share = "#{HOMEBREW_PREFIX}/share"
+        homebrew_in_xdg_data_dirs =
+          !ENV["XDG_DATA_DIRS"] || ENV["XDG_DATA_DIRS"] == "" ||
+          ENV["XDG_DATA_DIRS"].split(File::PATH_SEPARATOR).include?(share)
+        return if homebrew_in_xdg_data_dirs
+        <<~EOS.undent
+          Homebrew's share was not found in your XDG_DATA_DIRS but you have
+          this variable set to include other locations.
+          Some programs like `vapigen` may not work correctly.
+          Consider setting the XDG_DATA_DIRS for example like so
+              echo 'export XDG_DATA_DIRS="#{share}:$XDG_DATA_DIRS"' >> #{shell_profile}
+        EOS
+      end
+
+      def dyld_prefix
+        "LD"
+      end
+
+      def check_for_gettext; end
+      def check_for_iconv; end
+      def check_for_multiple_volumes; end
```

Comment:
You can delete these three lines. These methods are called by introspection. They won't be called if they don't exist.

Reply:
The idea was to override the default ones with empty ones on Linux where we don't need them...  please confirm.

Reply:
You can move these functions from generic to `os/mac/diagnostic.rb`, and `os/linux/diagnostic.rb` does not need any implementation.